pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2654,6 +2654,36 @@ func updatePortAndPciBackIoBundleAll(ctx *domainContext) {
 	}
 }
 
+// nvmeDeviceIsMountedByEVE checks if an NVME device is used by EVE-OS
+func nvmeDeviceIsMountedByEVE(pciID string) bool {
+	// Find /dev/nvmX from pciID
+	// e.g., ls /sys/bus/pci/devices/<pciID>/nvme/
+	//  -> nvme1
+	cmd := "/bin/ls"
+	cmdArgs := []string{"/sys/bus/pci/devices/" + pciID + "/nvme/"}
+	out, err := base.Exec(log, cmd, cmdArgs...).CombinedOutput()
+	if err != nil || out == nil {
+		log.Errorf("Could not find device with pciID %s, error:%+v, out:%s", pciID, err, string(out))
+		return false
+	}
+
+	// Extract nvmX from previous output
+	// e.g., out is /sys/class/nvme/nvme0/device/uevent:PCI_SLOT_NAME=0000:00:04.0
+	//  -> get nvme0 which is the 5th token of out using "/" delimiter
+	deviceName := strings.TrimSpace(string(out))
+	cmd = "/bin/grep"
+	cmdArgs = []string{deviceName, "/proc/mounts"}
+	out, err = base.Exec(log, cmd, cmdArgs...).CombinedOutput()
```

Comment:
This will have false matches if I for instance have a directory mount point called /tmp/doIhavenvmeHere.
The first string on the line needs to be /dev/nvmeX for it to be a match.

And I think it is better to not fork and exec another binary but instead just read /proc/mounts line by line comparing the prefix of the line using strings.HasPrefix().
(We've had issued under load that fork/exec gets very slow. Better to use golang code to do work.)

I briefly looked if there is an existing golang package which reads as parses /proc/mounts; if you find one (with a usable license) one can also consider using this. But here a Readline loop is sufficient so easy.

Reply:
Actually you should be able to use https://github.com/shirou/gopsutil/blob/434cc29594a81df897f1c07189309e6887bead42/disk/disk_linux.go#L260
(a package we already use in other places).

That function also handles /dev/root it seems.

HOWEVER, none of this handles zfs.
Need to check the drives which are being used by the zpool. The way to do that is to look at the ZFSPoolStatus published by zfsmanager to look for a matching DiskName.Name (/dev/nvmeX). However, zfsmanager is started after domainmgr, hence until it has published ZFSPoolStatus I think you need to have domainmgr treat nvme* as keepInHost and once ZFSPoolStatus is available you can determine which (if any) nvme drives are used by ZFS.

Reply:
Thanks for the pointers